dvc/stage.py

```diff
@@ -164,10 +167,15 @@ def __init__(
         self.outs = outs
         self.deps = deps
         self.md5 = md5
+        self.sha256 = sha256
         self.locked = locked
         self.tag = tag
         self._state = state or {}
 
+        self.hash = [self.PARAM_MD5]
+        if repo and repo.cache and repo.cache.local and repo.cache.local.hash:
```

Comment:
gain, this if looks a bit complicated to me. Are all of these ands are needed. If `yes` do we test different combinations?